listenbrainz/webserver/static/js/src/APIService.ts

```diff
@@ -133,13 +133,24 @@ export default class APIService {
     return parseInt(result.payload.count, 10);
   };
 
+  refreshYoutubeToken = async (): Promise<string> => {
+    return this.refreshAccessToken("youtube");
+  };
+
   refreshSpotifyToken = async (): Promise<string> => {
-    const response = await fetch("/profile/refresh-spotify-token", {
-      method: "POST",
-    });
+    return this.refreshAccessToken("spotify");
+  };
+
+  refreshAccessToken = async (service: string): Promise<string> => {
+    const response = await fetch(
+      `/profile/music-services/${service}/refresh/`,
```

Comment:
it's a bit annoying that the service name is in the middle of the url here - otherwise we could have passed the url as a page prop using url_for() to prevent errors in case this url changes